### PR TITLE
Zipping sources of endorsed libraries for IDE

### PR DESCRIPTION
**Pull request by LepilkinaElena**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -216,9 +216,39 @@ task distNativeSources(type: Zip) {
...
+    include('**/*.kt')
+
+    from(project(':endorsedLibraries').file('build/sources-common'))
+}
+
+task distEndorsedNativeSources(type: Zip) {
+    dependsOn ":endorsedLibraries:endorsedLibsNativeSources"
+
+    destinationDirectory = file("$distDir/sources")
+    archiveFileName = "kotlin-endorsed-native-sources.zip"
```
Comment (ddolovov) [inline build.gradle]: No, no. Please put the source code of every single library in a separate ZIP file. Please use pattern `LIBRARY-sources.zip` for the ZIP file name.

File: build.gradle [hunk 2]
```diff
@@ -216,9 +216,39 @@ task distNativeSources(type: Zip) {
...
+
+    includeEmptyDirs = false
+    include('**/*.kt')
+
+    from(project(':endorsedLibraries').file('build/sources-native'))
```
Comment (ddolovov) [inline build.gradle]: I think referring anything in `build` directory is a dangerous practice. You may refer Gradle tasks, source sets, etc - any entity that Gradle knows how to build and knows how to check if it should be invalidated and re-built. But `build` directory can be treated as always holding stale data.

Comment (ilmat192) [reply build.gradle]: > You may refer Gradle tasks, source sets, etc - any entity that Gradle knows how to build and knows how to check if it should be invalidated and re-built.

I think that using Gradle's dependency mechanism in the same manner as it's used for stdlib sources is the best choice. I've prototyped something similar for native sources: https://github.com/JetBrains/kotlin-native/commit/eb0f1b1489a479a93b534269e59659c35160129b.

Comment (LepilkinaElena) [reply build.gradle]: Thanks, but as Dmitry said that there should be separated zip archives for each library, there is no necessity in any coping, so I made all easier

**Pull request merged**